File: ANA SOFIA BARREA GIRALDO/EJERCICIOS3.JAVA/EjercicioP02.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class EjercicioP02 {
    public static void main(String[] args) {

        Scanner teclado = new Scanner(System.in);

        System.out.print("Ingrese un texto: ");
        String texto = teclado.nextLine().toLowerCase();

        HashMap<Character, Integer> frecuencia = new HashMap<>();

        for (char letra : texto.toCharArray()) {

            if (letra != ' ') {

                if (frecuencia.containsKey(letra)) {
                    frecuencia.put(letra, frecuencia.get(letra) + 1);
                } else {
                    frecuencia.put(letra, 1);
                }
            }
        }

        ArrayList<Map.Entry<Character, Integer>> resultado =
                new ArrayList<>(frecuencia.entrySet());

        Collections.sort(resultado,
                Comparator.comparing(Map.Entry<Character, Integer>::getValue).reversed());

        System.out.println("\nFrecuencia de las letras:");

        for (Map.Entry<Character, Integer> dato : resultado) {
            System.out.println(dato.getKey() + " : " + dato.getValue());
        }
    }
}

File: ANA SOFIA BARREA GIRALDO/EJERCICIOS2.CS/ejercicio_p02.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main()
    {
        Console.Write("Ingrese un texto: ");
        string texto = Console.ReadLine().ToLower();

        Dictionary<char, int> frecuencia = new Dictionary<char, int>();

        foreach (char letra in texto)
        {
            if (letra != ' ')
            {
                if (frecuencia.ContainsKey(letra))
                {
                    frecuencia[letra]++;
                }
                else
                {
                    frecuencia[letra] = 1;
                }
            }
        }

        var resultado = frecuencia.OrderByDescending(x => x.Value);

        Console.WriteLine("\nFrecuencia de las letras:");

        foreach (var dato in resultado)
        {
            Console.WriteLine(dato.Key + " : " + dato.Value);
        }
    }
}